packages/hub/src/components/LoadMoreViaSearchParam.tsx

```diff
@@ -0,0 +1,29 @@
+import { FC } from "react";
+
+import { useUpdateSearchParams } from "@/hooks/useUpdateSearchParams";
+
+import { LoadMore } from "./LoadMore";
+
+type Props = {
+  param?: string;
+};
+
+export const LoadMoreViaSearchParam: FC<Props> = ({ param = "limit" }) => {
+  const updateSearchParams = useUpdateSearchParams();
+
+  const action = (count: number) => {
+    updateSearchParams(
+      (params) => {
+        if (params.get(param)) {
+          const oldValue = parseInt(params.get(param) as string);
+          params.set(param, String(oldValue + count));
+        } else {
+          params.set(param, String(count));
+        }
+      },
+      { mode: "replace", scroll: false }
+    );
```

Comment:
This code is initially written for the case where we render the content based on query strings on server, in RSC.

In `/ai`, we simply pass the workflows to the client component, then use a react hook, and render only based on the data from that hook. So, for example,`scroll: false` doesn't do anything useful.

But this component is reusable - if we proceed with the plan from #3154, then we'll use it for all pagination, and it should work reliably everywhere, e.g. on the hub's front page.

PS: it's also possible to do an invisible `<AutoLoadMore />` that loads when it's scrolled into the view (that's how it worked in my side project that I took this code from), but in this case, workflows are kind of heavy and an explicit button click is probably better.